makefile

```diff
@@ -0,0 +1,37 @@
+SOURCE_DIR  := ./src
+INCLUDE_DIR := ./include
+BUILD_DIR   := ./build
+
+CC        := gcc
+CFLAGS    := -Wall
+LDFLAGS   := -lm
+INC_FLAGS := -I$(INCLUDE_DIR)
+
+DEPS = $(INCLUDE_DIR)/creator.h $(INCLUDE_DIR)/finalizer.h $(INCLUDE_DIR)/consumer.h $(INCLUDE_DIR)/producer.h $(INCLUDE_DIR)/buffer.h
+
+.PHONY: all
+all: $(BUILD_DIR) $(BUILD_DIR)/creator $(BUILD_DIR)/producer $(BUILD_DIR)/consumer $(BUILD_DIR)/finalizer
+
+$(BUILD_DIR)/%.o: $(SOURCE_DIR)/%.c $(DEPS)
```

Comment:
Please remove "$(DEPS)", in C usually one specifies dependencies between objects and sources, headers are just "a part of sources files".